src/bin/main.rs

```diff
@@ -63,34 +61,41 @@ fn main() {
         return;
     }
 
-    // DEFAULT: optimal image search
-    let mut range = matches.values_of("range").unwrap();
-    let min: u8 = range
-        .next()
-        .unwrap()
-        .parse()
-        .expect("min range is not a valid integer");
-    let max: u8 = range
-        .next()
-        .unwrap()
-        .parse()
-        .expect("max range is not a valid integer");
-    let threshold: f64 = matches
-        .value_of("threshold")
-        .unwrap()
-        .parse()
-        .expect("threshold is not a valid number");
-    let original_path = matches.value_of("original").unwrap();
+    // Optimal image search
+    if let Some(matches) = matches.subcommand_matches("search") {
+        let mut range = matches.values_of("range").unwrap();
+        let min: u8 = range
+            .next()
+            .unwrap()
+            .parse()
+            .expect("min range is not a valid integer");
+        let max: u8 = range
+            .next()
+            .unwrap()
+            .parse()
+            .expect("max range is not a valid integer");
+        let threshold: f64 = matches
+            .value_of("threshold")
+            .unwrap()
+            .parse()
+            .expect("threshold is not a valid number");
+        let original_path = matches.value_of("original").unwrap();
 
-    let mut search = Search::from_path(
-        Path::new(original_path),
-        SearchOptions {
-            threshold: threshold,
-            quality_range: (min, max),
-            formats: vec![ImageFormat::JPEG],
-        },
-    ).unwrap();
-    search.run();
+        let mut search = Search::from_path(
+            Path::new(original_path),
+            SearchOptions {
+                threshold: threshold,
+                quality_range: (min, max),
+                formats: vec![ImageFormat::JPEG],
+            },
+        ).unwrap();
+        search.run();
+
+        println!("Optimal Config: {:?}", search.get_result());
+        return
+    }
 
-    println!("Optimal Config: {:?}", search.get_result());
+    // Show usage by default
+    println!("{}", matches.usage());
+    println!("{}", matches.subcommand_matches("search").unwrap().usage());
```

Comment:
Do we need `search` usage if you already have `println!("{}", matches.usage());`?

Reply:
trying to figure it out, I would like to have 2 separate subcommands and let the toplevel command only show docs, otherwise the args of the top level will pollute the subcommand